activation/poetdb.go

```diff
@@ -54,43 +37,40 @@ func (db *PoetDb) AddPoetProof(proof types.PoetProof) error {
 		return fmt.Errorf("failed to marshal poet proof: %v", err)
 	}
 
-	if err := db.store.Put(proof.Root, poetProof.Bytes()); err != nil {
+	ref := sha256.Sum256(poetProof.Bytes())
+
+	batch := db.store.NewBatch()
+	if err := batch.Put(ref[:], poetProof.Bytes()); err != nil {
 		return fmt.Errorf("failed to store poet proof: %v", err)
 	}
-
-	if err := db.store.Put(makeKey(proof.PoetId, proof.RoundId), proof.Root); err != nil {
+	if err := batch.Put(makeKey(proof.PoetId, proof.RoundId), ref[:]); err != nil {
 		return fmt.Errorf("failed to store poet proof index entry: %v", err)
 	}
+	if err := batch.Write(); err != nil {
+		return fmt.Errorf("failed to store poet proof and index: %v", err)
+	}
 
 	return nil
 }
 
-func (db *PoetDb) GetPoetProofRoot(poetId []byte, round *types.PoetRound) ([]byte, error) {
-	poetRoot, err := db.store.Get(makeKey(poetId, round.Id))
+func (db *PoetDb) GetPoetProofRef(poetId []byte, round *types.PoetRound) ([]byte, error) {
+	poetRef, err := db.store.Get(makeKey(poetId, round.Id))
 	if err != nil {
 		return nil, err
 	}
-	return poetRoot, nil
+	return poetRef, nil
 }
 
-func (db *PoetDb) GetMembershipByPoetProofRoot(poetRoot []byte) (map[common.Hash]bool, error) {
-	poetProofBytes, err := db.store.Get(poetRoot)
+func (db *PoetDb) GetMembershipByPoetProofRef(poetRef []byte) (map[common.Hash]bool, error) {
+	poetProofBytes, err := db.store.Get(poetRef)
```

Comment:
Don't you think it is better to store the members separately from the poet proof (still in one batch), to allow reading only the members? Anyway, we can leave it as a possible optimization (as part of the ticket that you added yesterday)

Reply:
I think the ultimate solution will be to have all the data in one entry for sync and additionally an entry whose key is the PoET ref and whose value is a bloom filter for validation.

If this is where we're headed, the current implementation is closer (just need to add the bloom filter).

If you don't think the optimization will happen soon or we see that the current implementation affects performance in a way we don't want to live with even temporarily, I can do what you suggested.

Reply:
just to be accurate its not relevant from sync's point of view
the question is how often do you need the members only.